apps/showcase/src/components/training/code-editor-view/code-editor-view.component.ts

```diff
@@ -382,10 +411,32 @@ export class CodeEditorViewComponent implements OnDestroy {
     }
   }
 
+  public ngOnInit() {
+    const { cwd } = this.project();
+    const savedState = localStorage.getItem(cwd);
+    if (savedState) {
+      const modal = this.modalService.open(SaveCodeDialogComponent, { backdrop: 'static' });
+      void modal.result.then(async (positiveReply) => {
```

Comment:
Can we try to ask the question once we have the cwdTree to avoid blocking the user too long?

Reply:
actually, should we block the user at all? we just need to block the editor view, no need to prevent the navigation or any other action?
I was thinking about putting the prompt on the right side to block the editor view but keep the rest of the showcase available

Reply:
Why not, let's see what it could look like

Reply:
I updated the screenshot